src/main/java/com/ibm/disni/RdmaServerEndpoint.java

```diff
@@ -130,8 +130,12 @@ public C accept() throws IOException {
 				}
 			}
 			C endpoint = requested.poll();
-			logger.info("connect request received");
-			endpoint.accept();
+
+			// a null endpoint gets returned if listening endpoint closes
+			if (endpoint != null) {
+				logger.info("connect request received");
+				endpoint.accept();
+			}
 			return endpoint;
```

Comment:
Are we sure applications are handling the case correctly if null is returned here? Might as well throw an exception (Much like before)?